### DISCUSSION TITLE
Shopping list card: paste multiple lines

### DISCUSSION THREAD

**Discussion by DavidMStraub** [home-assistant/frontend#11272] [Other feature requests]
## The request

I like the shopping list integration/card, but in practice I don't use it a lot since it lacks one killer feature, in my view: to be able to post a multi-line list, such as the ones my wife sends me in WhatsApp:

```
beer
milk
diapers
bread
```

In Google Keep, this can be pasted and automatically turns into a 4-item list:
![shoppinglist](https://user-images.githubusercontent.com/10965193/85545258-14a63d80-b61c-11ea-8ce0-d387b0a877fa.gif)

The shopping list card instead turns it into a single item with breaks replaced by spaces.

## The alternatives

None I can think of.

## Additional information

Using 0.111.0.

**Comment by DavidMStraub** [comment]
I could try implementing this myself if someone gives me a :+1: that the following idea could work:
- replace the `paper-input` by a `paper-textarea` (same height, but preventing the browser from stripping newlines on paste)
- on change (i.e. paste), search for newlines. if none exist, use current logic. If any exist, add each line except the last as a new item (`_addItem`). Leave the last line in the textarea. This would treat each pasted newline as a return key event, which makes sense.

**Comment by zsarnett** [comment]
I like this idea.

**Comment by iantrich** [comment]
Seems weird that the last one wouldn't be added as well

**Comment by jenseo** [comment]
@DavidMStraub Did you ever get going on this or could I give it a go?

**Comment by DavidMStraub** [comment]
I never got around to it, would be awesome if you gave it a shot!

**Comment by jenseo** [comment]
@DavidMStraub Will do my best! :)
